gertjvr/serverless-plugin-simulate
# FIX: Issue with endpoint path matching

## What did you implement:

Closes #40 

## How did you implement it:

Added a step for non `{param+}` urls which just adds regex `^` and `$` so that the url must exactly match for that endpoint to be selected. The `{param}` and greedy `{param+}` functionality should remain unchanged.

## Other changes:

- Upgraded `jest` to version 19 to be able to use `expect.assertions(number)` and fixed tests which no longer worked
- Added `use strict` to `lib/services/index.js` because it uses `const, let, arrow functions etc.`


## How can we verify it:

Create a couple of functions and define in the following order in the serverless.yml
```
functions:
  subpathHandler:
    handler: src/subpath
    events:
      - http:
          path: test/path
          method: post
          cors: true

  fullpathHandler:
    handler: src/fullpath
    events:
      - http:
          path: test/path/full
          method: post
          cors: true
```

A HTTP request to `POST http://<host:path>/test/path/full` should go to the function `fullpathHandler` and not `subpathHandler`

I also added some testing around `serve.createContext` so that we know each type of request works. I wrote tests for the base situations but more work needs to be done to make sure that the functionality of API Gateway is fully known for mixtures of parameters and greedy parameters.

## Todos:



***Is this ready for review?:*** YES
